google_guest_agent/non_windows_accounts_integ_test.go

```diff
@@ -3,10 +3,62 @@
 package main
 
 import (
+	"fmt"
+	"os"
 	"os/exec"
+	"strings"
 	"testing"
 )
 
+const (
+	testUser           = "integration-test-user"
+	defaultgroupstring = "adm,dip,docker,lxd,plugdev,video,google-sudoers"
+)
+
+func TestCreateAndRemoveGoogleUser(t *testing.T) {
+	if exist, err := userExists(testUser); exist == true {
+		t.Fatalf("test user should not exist")
+	}
+	if err := createGoogleUser(testUser); err != nil {
+		t.Errorf("createGoogleUser failed creating test user")
+	}
+	if exist, err := userExists(testUser); exist != true || err != nil {
+		t.Errorf("test user should exist")
+	}
+	cmd := exec.Command("groups", testUser)
+	ret := runCmdOutput(cmd)
+	if ret.ExitCode() != 0 {
+		t.Errorf("failed looking up groups for user: stdout:%s stderr:%s", ret.Stdout(), ret.Stderr())
+	}
+	groups := strings.Split(strings.TrimSpace(strings.Split(ret.Stdout(), ":")[1]), " ")
+	expectedGroups := config.Section("Accounts").Key("groups").MustString(defaultgroupstring)
+	for _, group := range groups {
+		if !contains(group, strings.Split(expectedGroups, ",")) {
```

Comment:
you're unnecessarily calling split on every iteration. try this:
```suggestion
	expectedGroupString := config.Section("Accounts").Key("groups").MustString(defaultgroupstring)
	expectedGroups := strings.Split(expectedGroupString, ",")
	for _, group := range groups {
		if !contains(group, expectedGroups) {
```